apps/src/templates/instructions/TeacherFeedback.jsx

```diff
@@ -153,6 +153,10 @@ class TeacherFeedback extends Component {
     }
   };
 
+  componentWillUnmount() {
+    window.removeEventListener('beforeunload');
+  }
```

Comment:
with this change, will there be a situation where our 'beforeunload' handler still gets triggered? 

I'd expect this event to be added while there are pending changes and then removed when the changes are saved to the server. However, I don't have a lot of direct experience with handling this event so rather than prescribing a solution I'll just say please make sure you're getting these warnings at the times that you'd expect to.

Reply:
It still is triggering the warning when we would want it to. 

I did this in response to the[ react documentation](https://reactjs.org/docs/react-component.html#componentdidmount) saying: "This method (componentDidMount) is a good place to set up any subscriptions. If you do that, don’t forget to unsubscribe in componentWillUnmount()."